src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -383,6 +389,9 @@ private int countCurrentEC2Slaves(SlaveTemplate template) throws AmazonClientExc
             filters.add(new Filter("launch.image-id", values));
         }
 
+        // The instances must match the jenkins server url
+        filters.add(new Filter("tag:" + EC2Tag.TAG_NAME_JENKINS_SERVER_URL + "=" + jenkinsServerUrl));
```

Comment:
if user specified `EC2Tag.TAG_NAME_JENKINS_SERVER_URL` manually, this line will break instance cap calculation.